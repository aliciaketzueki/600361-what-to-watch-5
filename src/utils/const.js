export const MORE_LIKE_NUM = 4;
export const MAX_ACTORS_NUM = 4;
export const SECOND = 1000;

export const tabs = [{
  name: `Overview`,
  id: 0,
}, {
  name: `Details`,
  id: 1,
}, {
  name: `Reviews`,
  id: 2
}];

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};
