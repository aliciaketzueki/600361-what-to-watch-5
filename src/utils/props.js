import PropTypes from "prop-types";

export const validFilm = PropTypes.oneOfType([
  PropTypes.oneOf([null]).isRequired,
  PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    posterImage: PropTypes.string.isRequired,
    previewImage: PropTypes.string.isRequired,
    backgroundImage: PropTypes.string.isRequired,
    backgroundColor: PropTypes.string.isRequired,
    videoLink: PropTypes.string.isRequired,
    previewVideoLink: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    scoresCount: PropTypes.number.isRequired,
    director: PropTypes.string.isRequired,
    starring: PropTypes.arrayOf(PropTypes.string).isRequired,
    runTime: PropTypes.number.isRequired,
    genre: PropTypes.string.isRequired,
    released: PropTypes.number.isRequired,
    isFavorite: PropTypes.bool.isRequired
  }).isRequired
]);

export const validBool = PropTypes.bool.isRequired;
export const validShape = PropTypes.shape().isRequired;
export const validArrayOfShape = PropTypes.arrayOf(PropTypes.shape()).isRequired;
export const validNum = PropTypes.number.isRequired;
export const validArrayOfString = PropTypes.arrayOf(PropTypes.string).isRequired;
export const validFunc = PropTypes.func.isRequired;
export const validString = PropTypes.string.isRequired;
export const validOneOfType = PropTypes.oneOfType([
  PropTypes.func,
  PropTypes.shape({current: PropTypes.any})
]);
