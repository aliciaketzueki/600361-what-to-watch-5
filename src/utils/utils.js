import genres from "../mocks/genres";

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

const getRandom = (min, max) => {
  return Math.floor(Math.random() * (max - min) + min);
};

export const createFilmCard = (arr) => {
  const filmArr = [];

  [].forEach.call(arr, (name) => {
    const film = {};
    film.year = getRandom(1990, 2020);
    film.videoSrc = `https://docs.google.com/uc?id=1lOjED9R5dp-caVj1S8Wf8jjr3ob_qThW`;
    film.id = getRandom(1000, 9999);
    film.name = name;
    film.poster = film.name.split(` `).join(`-`).split(/[^a-zA-Z0-9_\-]/).join(``).toLowerCase() + `.jpg`;
    film.genreId = getRandom(0, 9);
    film.genre = genres[film.genreId].name;
    filmArr.push(film);
  });

  return filmArr;
};
