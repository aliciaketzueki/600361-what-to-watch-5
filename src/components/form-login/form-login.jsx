import React from "react";
import {validFunc, validString, validShape, validBool} from "../../utils/props";
import {connect} from "react-redux";
import {login, checkAuth} from "../../store/actions/api-actions";

const FormLogin = (props) => {
  const {email, password, emailValid, passwordValid, formErrors, formValid, handleSubmit, handleFieldChange, checkValid, onSubmit, isAuth} = props;

  return (
    <form action="#" className="sign-in__form" onSubmit={(e) => {
      handleSubmit(e);
      onSubmit({email, password});
    }}
    >
      {
        <div className="sign-in__message">
          {
            Object.keys(formErrors).map((fieldName, i) => {
              if (formErrors[fieldName].length > 0) {
                return (
                  <p key={i}>{formErrors[fieldName]}</p>
                );
              } else {
                return ``;
              }
            })
          }
          {/* {
            !isAuth() &&
            <p>We canâ€™t recognize this email and password combination. Please try again.</p>
          } */}
        </div>
      }

      <div className="sign-in__fields">
        <div className={`sign-in__field ${!emailValid ? `sign-in__field--error` : ``}`}>
          <input
            className="sign-in__input"
            type="email"
            placeholder="Email address"
            name="email"
            id="email"
            value={email}
            onChange={(e) => {
              handleFieldChange(e);
              checkValid(e);
            }}
          />
          <label className="sign-in__label visually-hidden" htmlFor="email">Email address</label>
        </div>
        <div className={`sign-in__field ${!passwordValid ? `sign-in__field--error` : ``}`}>
          <input
            className="sign-in__input"
            type="password"
            placeholder="Password"
            name="password"
            id="password"
            value={password}
            onChange={(e) => {
              handleFieldChange(e);
              checkValid(e);
            }}
          />
          <label className="sign-in__label visually-hidden" htmlFor="password">Password</label>
        </div>
      </div>
      <div className="sign-in__submit">
        <button
          className="sign-in__btn"
          type="submit"
          disabled={!formValid}
        >
          Sign in
        </button>
      </div>
    </form>
  );
};

const mapDispatchToProps = (dispatch) => ({
  isAuth() {
    dispatch(checkAuth());
  },
  onSubmit(authData) {
    dispatch(login(authData));
  }
});

FormLogin.propTypes = {
  onSubmit: validFunc,
  handleSubmit: validFunc,
  handleFieldChange: validFunc,
  checkValid: validFunc,
  email: validString,
  password: validString,
  formErrors: validShape,
  emailValid: validBool,
  passwordValid: validBool,
  formValid: validBool,
};

export {FormLogin};
export default connect(null, mapDispatchToProps)(FormLogin);
