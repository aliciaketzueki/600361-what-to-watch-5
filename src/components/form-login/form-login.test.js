import React from "react";
import renderer from "react-test-renderer";
import {FormLogin} from "./form-login";
import {Validation} from "../../utils/const";
const {PASSWORD, COMMENT_MAX, COMMENT_MIN} = Validation;

const noop = () => {};

const formErrors = {
  email: `Please enter a valid email address`,
  password: `Password must be at least ${PASSWORD} characters`,
  textReview: `Review text must be at least ${COMMENT_MIN} and no more than ${COMMENT_MAX} characters`,
};

describe(`Render FormLogin`, () => {
  test.each([
    [`with`, true],
    [`without`, false],
  ])(`%s validation`, (_expected, isValid) => {
    const tree = renderer
      .create(
          <FormLogin
            email={`aliciaketzueki@gmail.com`}
            password={`12345678990`}
            emailValid={true}
            passwordValid={true}
            formErrors={formErrors}
            formValid={isValid}
            handleSubmit={noop}
            handleFieldChange={noop}
            checkValid={noop}
            onSubmit={noop}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
