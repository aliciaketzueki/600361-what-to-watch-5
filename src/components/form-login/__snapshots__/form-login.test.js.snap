// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render FormLogin with validation 1`] = `
<form
  action="#"
  className="sign-in__form"
  onSubmit={[Function]}
>
  <div
    className="sign-in__message"
  >
    <p>
      Please enter a valid email address
    </p>
    <p>
      Password must be at least 6 characters
    </p>
    <p>
      Review text must be at least 50 and no more than 400 characters
    </p>
  </div>
  <div
    className="sign-in__fields"
  >
    <div
      className="sign-in__field "
    >
      <input
        className="sign-in__input"
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="Email address"
        type="email"
        value="aliciaketzueki@gmail.com"
      />
      <label
        className="sign-in__label visually-hidden"
        htmlFor="email"
      >
        Email address
      </label>
    </div>
    <div
      className="sign-in__field "
    >
      <input
        className="sign-in__input"
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value="12345678990"
      />
      <label
        className="sign-in__label visually-hidden"
        htmlFor="password"
      >
        Password
      </label>
    </div>
  </div>
  <div
    className="sign-in__submit"
  >
    <button
      className="sign-in__btn"
      disabled={false}
      type="submit"
    >
      Sign in
    </button>
  </div>
</form>
`;

exports[`Render FormLogin without validation 1`] = `
<form
  action="#"
  className="sign-in__form"
  onSubmit={[Function]}
>
  <div
    className="sign-in__message"
  >
    <p>
      Please enter a valid email address
    </p>
    <p>
      Password must be at least 6 characters
    </p>
    <p>
      Review text must be at least 50 and no more than 400 characters
    </p>
  </div>
  <div
    className="sign-in__fields"
  >
    <div
      className="sign-in__field "
    >
      <input
        className="sign-in__input"
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="Email address"
        type="email"
        value="aliciaketzueki@gmail.com"
      />
      <label
        className="sign-in__label visually-hidden"
        htmlFor="email"
      >
        Email address
      </label>
    </div>
    <div
      className="sign-in__field "
    >
      <input
        className="sign-in__input"
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value="12345678990"
      />
      <label
        className="sign-in__label visually-hidden"
        htmlFor="password"
      >
        Password
      </label>
    </div>
  </div>
  <div
    className="sign-in__submit"
  >
    <button
      className="sign-in__btn"
      disabled={true}
      type="submit"
    >
      Sign in
    </button>
  </div>
</form>
`;
